lambda/repository/lambda_functions.py

```diff
@@ -128,14 +137,47 @@ def similarity_search(event: dict, context: dict) -> Dict[str, Any]:
         ValidationError: If required parameters are missing or invalid
     """
     query_string_params = event["queryStringParameters"]
-    model_name = query_string_params["modelName"]
     query = query_string_params["query"]
     top_k = query_string_params.get("topK", 3)
     include_score = query_string_params.get("score", "false").lower() == "true"
     repository_id = event["pathParameters"]["repositoryId"]
+    collection_id = query_string_params.get("collectionId")
 
     repository = get_repository(event, repository_id=repository_id)
 
+    # Get user context for collection access
+    username = get_username(event)
```

Comment:
We do these 3 things **so** many places, should this maybe just be a helper function that returns a tuple?